web/modules/custom/custom_weather/src/Form/CustomWeatherUserForm.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Drupal\custom_weather\Form;
+
+use Drupal\Core\Database\Database;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\custom_weather\Service\UserCityHandler;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a configuration form for managing settings related to the module.
+ */
+class CustomWeatherUserForm extends FormBase {
+
+  public function __construct(protected UserCityHandler $city) {
+    $this->databaseService = $city;
```

Comment:
Please read about constructor promotion again and fix this
(databaseService is misleading name as well)